### ISSUE TITLE
Widget sizing issues.

### ISSUE THREAD

**Issue by jtappin** [gnudatalanguage/gdl#1470]
This is just an initial heads-up as I've not yet made a demo routine to show the problem. 

Since Manjaro dropped support for WxWidgets gtk-2, there have been several problems with widget bases being too small to show all of the widgets contained within. The biggest problem seems to be when there are tab widgets (but not exclusively).

At present the two routines that are showing the problem are probably too complex to be useful as test beds (graffer and also a save file explorer that uses the full capabilities of Craig Markwardt's CMS package).

The graffer case is accompanied with lots of messages like:
````
(gdl:1094233): Gtk-CRITICAL **: 10:40:49.327: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkNotebook
````
No error messages from `examine_save` which does not have tab widgets.

Will post demos when I have something useful.

GDL from GIT to commit: https://github.com/gnudatalanguage/gdl/commit/0629d02a92fb6b497e17bb674fa8ea76601a6ae8
WX Widgets: wxwidgets-common 3.2.1-1, wxwidgets-gtk3 3.2.1-1

**Comment by jtappin** [comment]
Here is  quick and dirty demo. Can be run with or without the `/subtab`  keyword.

On my system, without subtab then the second tab only shows six of seven text widgets.
With subtab, then on the first tab, the first subtab only shows 4 1/2 lines and the second only 3 1/2 buttons.
In neither case can the `quit` button be accessed.

[tab_test.pro](https://github.com/gnudatalanguage/gdl/files/10392098/tab_test.pro.txt)

**Comment by GillesDuvert** [comment]
Thanks, this was hanging around from som time here.
GTK is very fussy and such message annoy a lot of people using a lot of applications, as one can easily check on the web.
Perhaps there is a way to make GTK silent.

**Comment by GillesDuvert** [comment]
Concerning the tab sizing, this is a bug and your example is very helpful.

**Comment by jtappin** [comment]
The fact that there wasn't a problem with WXWidgets-gtk2 may also be useful.
Tabs aren't the only widgets with sizing problems, but they are the worst. The other case I have, I've not yet been able to isolate it in a code that doesn't depend on quite a few extra routines that are not part of the  GDL distribution.

**Comment by GillesDuvert** [comment]
widget sizing with wxWidgets has been THE problem from the start, because the process of widget creation in GDL is not synchronous with the real creation in wxWidgets, itself with sizes and default sizes depending on the underlying 'real' widget library (GTK...) . All of this construct is very fragile indeed.

**Comment by GillesDuvert** [comment]
Annoying GTK messages will be supressed by #1472.
Widget sizing will be examined later.

**Comment by GillesDuvert** [comment]
Fixed in #1483
